### ISSUE TITLE
Location sensitive attention shouldn't use preprocessed memory? 

### ISSUE THREAD

**Issue by rafaelvalle** [keithito/tacotron#117]
Keith,
I see that you're also using the `keys` (preprocessed memory) to compute the location sensitive attention scores. Shouldn't it be just the processed query and the processed location?

https://github.com/keithito/tacotron/blob/c94ab2757d52e4294dcd6a8da03f49d251b2dec4/models/attention.py#L61

**Comment by keithito** [comment]
I think "keys" in the AttentionWrapper terminology is the Ws_{i-1} term in equation 9 of https://arxiv.org/pdf/1506.07503.pdf

But I could be wrong. I'll take another look at the [docs](https://www.tensorflow.org/versions/master/tutorials/seq2seq) in a bit.

**Comment by rafaelvalle** [comment]
In the paper Ws_{i-1} is the hidden state of the recurrent neural network, Wh_j are the preprocessed encoder outputs and W\alpha_{i-1} are the attention weights or alignments.
On TensorFlow's seq2seq and @hirofumi0810 implementation of _End to End Speaker Recognition_, keys are the encoder outputs.

https://github.com/hirofumi0810/tensorflow_end2end_speech_recognition/blob/65b9728089d5e92b25b92384a67419d970399a64/models/attention/decoders/attention_layer.py#L151

**Comment by rafaelvalle** [comment]
@keithito FYI: I tried removing the variable but tf complained it can't be removed, I don't remember the error. I then simply multiplied it by 0 to cancel it's contribution to the gradients, etc...

**Comment by rafaelvalle** [comment]
@keithito I looked at the paper again and think I understand what's happening:
In the paper they say _We use the location sensitive attention from [21], which extends the additive attention mechanism [22] to use cumulative attention weights from previous decoder time steps
as an additional feature_. What they are using is not the `location based attention` but the `hybrid` location that _extends_ the content based attention with location awareness. 

Although the name is wrong, i.e. should be hybrid instead of location, the code you wrote is for the hybrid attention and it happens to match what's not so clearly described on the paper.

**Comment by keithito** [comment]
Yes -- sorry for the confusion. I'll update the naming and comments to be more clear about this.
